homeassistant/components/notify/group.py

```diff
@@ -0,0 +1,55 @@
+"""
+Group platform for notify component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.group/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (CONF_PLATFORM, CONF_NAME,
+                                 CONF_ENTITY_ID)
+from homeassistant.components.notify import (DOMAIN, ATTR_MESSAGE, ATTR_DATA,
+                                             BaseNotificationService)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ENTITIES = "entities"
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): "group",
+    vol.Optional(CONF_NAME): vol.Coerce(str),
+    vol.Required(CONF_ENTITIES): vol.All(cv.ensure_list, [{
+        vol.Optional(CONF_ENTITY_ID): vol.Any(cv.string, None),
```

Comment:
@balloob Suggestion on what to call it if not `entity_id`? I agree though that we shouldn't confuse the terminology.

`target` can still be passed, but it's in the `data` dictionary now. I did it that way so that you could pass whatever you wanted into the platform for the target instead of only the target, i.e. 

```yaml
- platform: group
    name: RobbiesDevices
    entities:
      - entity_id: html5
        data:
          target: "unnamed device"
      - entity_id: robbieiphone
        data:
          data:
            push:
              sound: US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav
      - entity_id: robbieipad
        data:
          data:
            push:
              sound: US-EN-Alexa-Mail-Has-Arrived.wav
```